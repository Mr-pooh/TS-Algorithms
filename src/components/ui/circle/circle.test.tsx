import { render, screen } from "@testing-library/react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe('Circle', () => {
    it('render without letter', () => {
        render(<Circle />);
        const circle = screen.getByTestId('circle-test');
        expect(circle).toMatchSnapshot();
    });

    it('render with letter', () => {
        render(<Circle letter="3" />);
        const circle = screen.getByTestId('circle-test');
        expect(circle).toMatchSnapshot();
    });
    it('render with head', () => {
        render(<Circle head={'s'} />)
        const circle = screen.getByTestId('circle-test');
        expect(circle).toMatchSnapshot()
    })
    it('render element-react with head', ()=> {
        render(<Circle head={<div><p>sss1123</p></div>} />)
        const circle = screen.getByTestId('circle-test');
        expect(circle).toMatchSnapshot()
    })
    it('render with tail', ()=> {
        render(<Circle tail={'S'} />)
        const circle = screen.getByTestId('circle-test')
        expect(circle).toMatchSnapshot()
    })
    it('render element-react with tail', ()=> {
        render(<Circle tail={<div><p>sss1123</p></div>} />)
        const circle = screen.getByTestId('circle-test')
        expect(circle).toMatchSnapshot()
    })
    it('render with index', ()=> {
        render(<Circle index={2} />)
        const circle = screen.getByTestId('circle-test')
        expect(circle).toMatchSnapshot()
    })
    it('render with prop isSmall true', ()=> {
        render(<Circle isSmall={true} />)
        const circle = screen.getByTestId('circle-test')
        expect(circle).toMatchSnapshot()
    })
    it('render state default', ()=> {
        render(<Circle state={ElementStates.Default} />)
        const circle = screen.getByTestId('circle-test')
        expect(circle).toMatchSnapshot()
    })
    it('render state changing', ()=> {
        render(<Circle state={ElementStates.Changing} />)
        const circle = screen.getByTestId('circle-test')
        expect(circle).toMatchSnapshot()
    })
    it('render state modified', ()=> {
        render(<Circle state={ElementStates.Modified} />)
        const circle = screen.getByTestId('circle-test')
        expect(circle).toMatchSnapshot()
    })
});