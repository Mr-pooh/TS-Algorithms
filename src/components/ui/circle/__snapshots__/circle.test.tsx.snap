// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle render element-react with head 1`] = `
<div
  class="content "
  data-testid="circle-test"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  >
    <div>
      <p>
        sss1123
      </p>
    </div>
  </div>
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle render element-react with tail 1`] = `
<div
  class="content "
  data-testid="circle-test"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  >
    <div>
      <p>
        sss1123
      </p>
    </div>
  </div>
</div>
`;

exports[`Circle render state changing 1`] = `
<div
  class="content "
  data-testid="circle-test"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    class="circle   changing"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle render state default 1`] = `
<div
  class="content "
  data-testid="circle-test"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle render state modified 1`] = `
<div
  class="content "
  data-testid="circle-test"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    class="circle   modified"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle render with head 1`] = `
<div
  class="content "
  data-testid="circle-test"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="circleHead"
  >
    s
  </div>
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle render with index 1`] = `
<div
  class="content "
  data-testid="circle-test"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  >
    2
  </p>
  <div
    class="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle render with letter 1`] = `
<div
  class="content "
  data-testid="circle-test"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    >
      3
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle render with prop isSmall true 1`] = `
<div
  class="content "
  data-testid="circle-test"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    class="circle  small default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;

exports[`Circle render with tail 1`] = `
<div
  class="content "
  data-testid="circle-test"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="circleTail"
  >
    S
  </div>
</div>
`;

exports[`Circle render without letter 1`] = `
<div
  class="content "
  data-testid="circle-test"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circleHead"
  />
  <div
    class="circle   default"
    data-testid="circle"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circleLetter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circleTail"
  />
</div>
`;
